src/include/main/client_context.h

```diff
@@ -55,6 +55,14 @@ class ClientContext {
 
     inline bool isTimeOutEnabled() const { return timeoutInMS != 0; }
 
+    inline uint64_t getTimeoutRemainingInMS() {
+        if (!isTimeOutEnabled()) {
+            return UINT64_MAX;
+        }
```

Comment:
We could change this into a `KU_ASSERT()`... not sure if that makes more sense. I.e., make it an error to call this method when a timeout is not set.